src/project/controller/item.ts

```diff
@@ -19,30 +19,34 @@ class Project extends Controller {
   }
 
   async put(ctx: Context) {
-    const project = await projectService.findById(
-      +ctx.params.projectId,
-    );
+    const project = await projectService.findById(+ctx.params.projectId);
 
-    ctx.request.validate<ProjectSchema>('https://tt.badgateway.net/schema/project.json');
+    ctx.request.validate<ProjectSchema>(
+      'https://tt.badgateway.net/schema/project.json'
+    );
     const body = ctx.request.body;
 
     const clientUrl = ctx.request.links.get('client');
     if (clientUrl) {
-      const clientId = (clientUrl.href.split('/').pop());
+      const clientId = clientUrl.href.split('/').pop();
       if (!clientId) {
-        throw new BadRequest('The client link must be in the format /clients/123');
+        throw new BadRequest(
+          'The client link must be in the format /client/123'
+        );
       }
 
       const client = await clientService.findById(+clientId);
       project.client = client;
-
     }
 
     project.name = body.name;
     await projectService.update(project);
 
     ctx.status = 204;
-
+    ctx.response.links.add({
+      rel: 'invalidates',
+      href: '/project',
+    });
```

Comment:
Just curious, why was this needed?

Reply:
Great question!
When updating a project (like updating the associated client) the `/project` list of projects collection wouldn't show the updated client, it would just be blank,
I tried adding the option `const {items} = useCollection(props.resource, {refreshOnStale: true});` to the CollectionLinks, where we're getting the list of projects, but it didn't resolve the issue.
The only ways I could remember to get resources/collection to update properly when another is changed is to pass the option `refreshOnStale`, or to add a link to whatever request to invalidate whatever is dependant on that updated info. Since refreshOnStale didn't work, I went with the invalidation links, which worked out. 🤷

Reply:
Does this have to do with fetching the title of the project? This makes sense to me if we fetch a list of projects (but not the children) so titles are just extracted from the `item` link.

Reply:
Yeah! You got it!